packages/dht/src/helpers/PeerID.ts

```diff
@@ -0,0 +1,94 @@
+import { UUID } from "./UUID"
+import { IllegalArguments } from './errors'
+import crypto from "crypto"
+
+export type PeerIDKey = string & { readonly __brand: 'peerIDKey' } // Nominal typing 
```

Comment:
```suggestion
export type PeerIDKey = BrandedString<'PeerIDKey'> 
```

(see e.g. `EthereumAddress.ts` in `utils`)